src/main/java/net/mcreator/workspace/misc/CreativeTabsOrder.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.workspace.misc;
+
+import net.mcreator.element.GeneratableElement;
+import net.mcreator.element.parts.TabEntry;
+import net.mcreator.element.types.interfaces.ITabContainedElement;
+import net.mcreator.workspace.elements.ModElement;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CreativeTabsOrder extends ConcurrentHashMap<String, ArrayList<String>> {
+
+	public void addModElementToTabs(ModElement element) {

```

Comment:
Is this one really needed?

updateModElementTabs should do the same thing, right?

The thing is when calling addModElementToTabs, "element.getGeneratableElement()" may even return null or introduce circular calls

Reply:
Is this still a problem?

Reply:
Now it should be fully resolved